app/gui/integration-test/dashboard/actions/api.ts

```diff
@@ -23,6 +23,16 @@ import invariant from 'tiny-invariant'
 // === Constants ===
 // =================
 
+let lastTimestamp = 0
+
+/** Return a new date that has not yet been returned. */
+function newUniqueDate(): Date {
+  const timestamp = Number(new Date())
```

Comment:
Why's it that complex? The time always change and a new date is always unique, isn't it?

Reply:
`newUniqueDate(); newUniqueDate()` is going to often return the same date.

i'm not 100% sure it fixed anything but i *think* i was getting flakiness without this - note that the specific problem this was causing issues was `setupAPI` which is synchronous

Reply:
agree it's quite complex though

Reply:
I was thinking on just doing an arbitrary delay for a call with an empty while loop. It'd be slower, but the diff will always be at leats 1ms